op-e2e/custom_gas_token_test.go

```diff
@@ -0,0 +1,456 @@
+package op_e2e
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/receipts"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCustomGasToken(t *testing.T) {
+	InitParallel(t, SkipOnFPAC) // Custom Gas Token feature is not yet compatible with FPAC
+
+	cfg := DefaultSystemConfig(t)
+	cfg.DeployConfig.FinalizationPeriodSeconds = 2 // 2s finalization period
+
+	sys, err := cfg.Start(t)
+	require.NoError(t, err, "Error starting up system")
+	defer sys.Close()
+
+	log := testlog.Logger(t, log.LevelInfo)
+	log.Info("genesis", "l2", sys.RollupConfig.Genesis.L2, "l1", sys.RollupConfig.Genesis.L1, "l2_time", sys.RollupConfig.Genesis.L2Time)
+
+	l1Client := sys.Clients["l1"]
+	l2Client := sys.Clients["sequencer"]
+
+	aliceOpts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Alice, cfg.L1ChainIDBig())
+	require.NoError(t, err)
+
+	// Deploy WETH9, we'll use this as our custom gas token for the purpose of the test
+	weth9Address, tx, weth9, err := bindings.DeployWETH9(aliceOpts, l1Client)
+	require.NoError(t, err)
+	_, err = wait.ForReceiptOK(context.Background(), l1Client, tx.Hash())
+	require.NoError(t, err)
+
+	// setup expectations using custom gas token
+	type Expectations struct {
+		tokenAddress  common.Address
+		tokenName     string
+		tokenSymbol   string
+		tokenDecimals uint8
+	}
+	disabledExpectations := Expectations{
+		common.HexToAddress("0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"),
+		"Ether",
+		"ETH",
+		uint8(0x12),
```

Comment:
why use hex here?

Reply:
No particular reason, I think when this gets logged out it looks this way. Reverted to decimal here [073b51b](https://github.com/ethereum-optimism/optimism/pull/10440/commits/073b51b0617c1ffd9a4fa6519e228b2818b071c6)